src/main/java/com/academia/library/mapper/BookMapper.java

```diff
@@ -2,16 +2,50 @@
 
 import com.academia.library.dto.BookRequestDto;
 import com.academia.library.dto.BookResponseDto;
+import com.academia.library.model.Author;
 import com.academia.library.model.Book;
+import com.academia.library.model.Tag;
+import com.academia.library.repository.AuthorRepository;
+import com.academia.library.repository.TagRepository;
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.springframework.beans.factory.annotation.Autowired;
 
-@Mapper(componentModel = "spring")
-public interface BookMapper {
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Mapper(componentModel = "spring",
+        imports = {LocalDateTime.class})
+public abstract class BookMapper {
+
+    @Autowired
+    private AuthorRepository authorRepository;
+
+    @Autowired
+    private TagRepository tagRepository;
+
+    @AfterMapping
```

Comment:
If you use AfterMapping please use it after base mapping method

Reply:
Ok, thank